compiler/noirc_frontend/src/hir_def/types/arithmetic.rs

```diff
@@ -215,8 +223,12 @@ impl Type {
                 if l_op == Division {
                     op = op.inverse()?;
                 }
+
+                let divides_evenly = !lhs.infix_kind(rhs).is_field_element()
+                    && l_const.to_i128() % r_const.to_i128() != 0;
```

Comment:
I think this is backwards. If it divides evenly the remainder should be 0.

We'd also need to check for divide by zero before this

Reply:
Updated negation and checking division by zero using `checked_rem`